src/analyzer/protocol/ssl/spicy/SSL.spicy

```diff
@@ -0,0 +1,2058 @@
+
+    on %done {
+        msg.first_packet = False;
+    }
+
+    on %init {
+        if (sh.negotiated_version != UNKNOWN_VERSION && sh.negotiated_version != SSLv2) {
+            spicy::decline_input("Late switch to SSLv2 record");
+        }
+    }
+};
+
+type SSL2ClientMasterKey = unit(length: uint16, inout sh: Share) {
+    : skip bytes &size=length;
+
+    on %init {
+        sh.client_encrypted = True;
+        sh.server_encrypted = True;
+        sh.established = True;
+        zeek::skip_input();
+    }
+};
+
+# For TLS-y protocols - determine how to continue
+type TLSRecordFragmentChoice = unit(content_type: uint8, handshakesink: sink&, alertsink: sink&, inout msg: Message, inout sh: Share) {
+    # content_type: uint8; # &convert=ContentType($$);
+    version: uint16;
+
+    switch (is_dtls_version(self.version)) {
+        True -> dtlsfragment: DTLSRecordFragment(content_type, handshakesink, alertsink, msg, sh);
+        False -> tlsfragment: TLSRecordFragment(content_type, handshakesink, alertsink, msg, sh);
+    };
+
+    on version {
+        msg.record_version = self.version;
+        if (is_dtls_version(self.version)) {
+            msg.dtls = True;
+        }
+    }
+};
+
+type TLSRecordFragment = unit(content_type: uint8, handshakesink: sink&, alertsink: sink&, inout msg: Message, inout sh: Share) {
+    record: PlaintextRecord(content_type, handshakesink, alertsink, msg, sh);
+};
+
+type DTLSRecordFragment = unit(content_type: uint8, handshakesink: sink&, alertsink: sink&, inout msg: Message, inout sh: Share) {
+    # the epoch signalizes that a changecipherspec message has been received. Hence, everything with
+    # an epoch > 0 should be encrypted
+    epoch: uint16;
+    sequence_number: bytes &size=6; # uint48
+    record: PlaintextRecord(content_type, handshakesink, alertsink, msg, sh);
+};
+
+type PlaintextRecord = unit(content_type: uint8, handshakesink: sink&, alertsink: sink&, inout msg: Message, inout sh: Share) {
+    length: uint16;
+    var encrypted: bool;
+    # convenient triggers to hang stuff in the evt file from. Two of them for event ordering :)
+    trigger_zero: void;
+    trigger_one: void;
+    switch (ContentType(content_type)) {
+        ContentType::handshake -> : bytes &size=self.length -> handshakesink;
+        ContentType::application_data -> {
+            switch (self.encrypted) {
+                False -> appdata: bytes &size=self.length;
+                True -> : skip bytes &size=self.length;
+            };
+        }
+        ContentType::change_cipher_spec -> ccs: bytes &size=self.length;
+        ContentType::heartbeat -> hn: Heartbeat(sh, self.length);
+        ContentType::alert -> : bytes &size=self.length -> alertsink;
+        * -> unhandled: skip bytes &size=self.length;
+    };
+    trigger_two: void;
+
+    on length {
+        self.encrypted = determine_encryption_on(self, content_type, handshakesink, alertsink, sh);
```

Comment:
This `self.encrypted` is used on 792 above? Similar as for other comments, any reason not do add it directly as:
```
var encrypted: bool;
length: uint16  {
    self.encrypted = determine_encryption_on(self, content_type, handshakesink, alertsink, sh);
}
```